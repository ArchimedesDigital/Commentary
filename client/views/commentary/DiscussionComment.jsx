import IconButton from 'material-ui/IconButton';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import FontIcon from 'material-ui/FontIcon';

DiscussionComment = React.createClass({

  propTypes: {
    discussionComment: React.PropTypes.object.isRequired,
		currentUser: React.PropTypes.object
  },

  getInitialState(){
    return {
			editMode: false
		}

  },

	showEditMode(){
		this.setState({
			editMode: true
		});

	},

	closeEditMode(){
		this.setState({
			editMode: false
		});

	},

	updateDiscussionComment(){

		var content = $(this.refs.updateCommentForm).find("textarea").val();

		Meteor.call("discussionComments.update", {
			_id: this.props.discussionComment._id,
			content: content
		});

		this.setState({
			editMode: false
		});

	},

	upvoteDiscussionComment(){
		if(typeof this.props.currentUser !== "undefined"){
			Meteor.call("discussionComments.upvote",
				this.props.discussionComment._id
			);

		}

	},

  render() {
		var self = this;
    var userIsLoggedIn = this.props.currentUser ? true : false;
		var discussionComment = this.props.discussionComment;
		discussionComment.children = [];
		var userUpvoted = false;

		var username = "";

		if(discussionComment.user.emails.length){
			username = discussionComment.user.emails[0].address.split("@")[0];
		}

		if(typeof this.props.currentUser !== "undefined" && discussionComment.voters.indexOf(this.props.currentUser._id) >= 0){
				userUpvoted = true;
		}

    return (<div
							className="discussion-comment paper-shadow"
							>
                <div
									className="inner-comment-row">

                    <div className="discussion-commenter-profile-picture profile-picture paper-shadow">
                        <img src="/images/default_user.jpg"/>
                    </div>

                    <div className="discussion-commenter-meta">
                        <span className="discussion-commenter-name">
                            {username}
                        </span>
                        <span className="discussion-comment-date">
                            <span >Updated: </span>
														{moment(discussionComment.updated).format('D MMMM YYYY') || moment(discussionComment.created).format('D MMMM YYYY')}
                        </span>
                    </div>

                </div>
                <div className="inner-comment-row">
                    <div className="discussion-comment-text">

                        {/*<div
													dangerouslySetInnerHTML={{ __html: discussionComment.content}}
													></div>*/}
												{this.state.editMode ?
                          <form className="update-comment-form clearfix" name="update-comment-form" ref="updateCommentForm">
                            <textarea className="new-comment-text" defaultValue={this.props.discussionComment.content}></textarea>
														<div className="comment-edit-buttons">
	                            <RaisedButton
																label="Update"
																className="submit-comment-button paper-shadow"
																onClick={this.updateDiscussionComment}
																>
	                            </RaisedButton>
	                            <FlatButton
																label="Close "
	                              className="close-form-button "
																onClick={this.closeEditMode}>
	                            </FlatButton>
														</div>
                          </form>
												:
												<div>{discussionComment.content}</div>

											}

                    </div>
                </div>
                <div className="inner-comment-row" >
                    <FlatButton
											label={discussionComment.votes}
											onClick={this.upvoteDiscussionComment}
                      className={(userUpvoted) ? "vote-up upvoted" : "vote-up"}
		                  icon={<FontIcon className="mdi mdi-chevron-up" />}
											>
											{userIsLoggedIn ?
												""
												:
												<span className="vote-up-tooltip">
													You must be signed in to upvote.
												</span>
											}
                    </FlatButton>
                    {/*<FlatButton
											label="Reply"
											onClick={this.showReplyForm}
											className="reply">
                    </FlatButton>*/}
                    {/*<!--RaisedButton label="Share" className="share" onClick="share_discussionComment($event)">
                        Share
                    </RaisedButton-->*/}
										{("currentUser" in self.props && self.props.currentUser !== null && typeof self.props.currentUser !== "undefined")
											? (self.props.currentUser._id === discussionComment.user._id) ?
	                      <FlatButton
													label="Edit"
													onClick={this.showEditMode}
													className="edit"
													>
	                      </FlatButton>
												: ""
											: ""
										}
                    {/*<FlatButton
											label="Remove"
											onClick={this.removeDiscussionComment}
											className="remove"
											>
                    </FlatButton>*/}
                </div>


								{false ?
                  <div className="reply-create-form" >
                      <div className="add-comment-wrap">
                          <form className="new-comment-form" name="new-comment-form" >
                              <div className="add-comment-row-1" >
                                  {/*<!--div className="profile-picture paper-shadow">
                                      <asset:image src="default_user.jpg"/>
                                  </div-->*/}
                                  <textarea className="new-comment-text" placeholder="Enter your reply here . . . " enter-submit="add_discussion_reply($event)">
                                  </textarea>
                                  <RaisedButton label="Submit" type="submit" className="submit-comment-button paper-shadow" >Submit</RaisedButton>
                                  <RaisedButton label="Close Reply"
                                             className="close-form-button " onClick={this.closeReply}>Close
                                  </RaisedButton>
                              </div>
                          </form>
                      </div>
                  </div>
								: ""}

                <div className="discussion-comment-children">

									{discussionComment.children.map(function(discussionCommentChild, j){
                    <div
											key={j}
											className="discussion-comment discussion-comment-child"
											 >
                        <div className="inner-comment-row">
                            <div className="discussion-commenter-profile-picture profile-picture paper-shadow">
                                <img src="/images/default_user.png"/>
                            </div>
                            <div className="discussion-commenter-meta">
                                <span className="discussion-commenter-name">
                                    {discussionCommentChild.user.name}
                                </span>
                                <span className="discussion-comment-date">
                                    {discussionCommentChild.updated}
                                </span>
                            </div>

                        </div>
                        <div className="inner-comment-row">
                            <div className="discussion-comment-text">
                                <p dangerouslySetInnerHTML={{ __html: discussionCommentChild.content}}></p>
                            </div>
                        </div>
                        <div className="inner-comment-row">
	                        <FlatButton
														label={discussionComment.votes}
														onClick={this.upvoteDiscussionComment}
	                          className="vote-up upvoted"
					                  icon={<FontIcon className="mdi mdi-chevron-up" />}
														>
	                        </FlatButton>
	                        <FlatButton
														label="Reply"
														onClick={this.showReplyForm}
														className="reply">
	                        </FlatButton>
	                        {/*<!--RaisedButton label="Share" className="share" onClick="share_discussionComment($event)">
	                            Share
	                        </RaisedButton-->*/}
	                        <FlatButton
														label="Edit"
														onClick={this.editDiscussionComment}
														className="edit"
														>
	                        </FlatButton>
	                        <FlatButton
														label="Remove"
														onClick={this.removeDiscussionComment}
														className="remove"
														>
	                        </FlatButton>

                        </div>



                    {/*<!-- .discussion-comment-child -->*/}</div>

									})}

                {/*<!-- .discussion-comment-children -->*/}</div>

            {/*<!-- .discussion-comment -->*/}</div>

     );
   }

});
