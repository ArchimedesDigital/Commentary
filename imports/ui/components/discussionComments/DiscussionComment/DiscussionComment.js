import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Meteor } from 'meteor/meteor';
import { Session } from 'meteor/session';
import { createContainer } from 'meteor/react-meteor-data';
import { moment } from 'meteor/momentjs:moment';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import FontIcon from 'material-ui/FontIcon';
import { Link } from 'react-router-dom';
import { compose } from 'react-apollo';
import { 
	discussionCommentUpdateMutation, 
	discussionCommentReportMutation,
	discussionCommentsQuery,
	discussionCommentUpvoteMutation,
	discussionCommentUnreportMutation} 
	from '/imports/graphql/methods/discussionComments';

class DiscussionComment extends Component {

	constructor(props) {
		super(props);
		this.state = {
			editMode: false,
			moreOptionsVisible: false,
			shareOptionsVisible: false,
			readComment: false,
		};
		this.showEditMode = this.showEditMode.bind(this);
		this.closeEditMode = this.closeEditMode.bind(this);
		this.updateDiscussionComment = this.updateDiscussionComment.bind(this);
		this.upvoteDiscussionComment = this.upvoteDiscussionComment.bind(this);
		this.reportDiscussionComment = this.reportDiscussionComment.bind(this);
		this.unreportDiscussionComment = this.unreportDiscussionComment.bind(this);
		this.toggleMoreOptions = this.toggleMoreOptions.bind(this);
		this.toggleShareOptions = this.toggleShareOptions.bind(this);
		this.readDiscussionComment = this.readDiscussionComment.bind(this);
	}
	showEditMode() {
		this.setState({
			editMode: true,
		});
	}
	closeEditMode() {
		this.setState({
			editMode: false,
		});
	}
	updateDiscussionComment() {
		const content = $(this.updateCommentForm).find('textarea').val();
		const { discussionComment } = this.props;
		this.props.discussionCommentUpdate(discussionComment._id, content).catch((e) => {
			console.log(e);
		});
		this.setState({
			editMode: false,
		});
	}
	upvoteDiscussionComment() {
		const { currentUser, discussionComment } = this.props;
		if (currentUser) {
			this.props.discussionCommentUpvote(discussionComment._id).catch((e) => {
				console.log(e);
			});
		}
	}
	reportDiscussionComment() {
		const { currentUser, discussionComment } = this.props;
		if (currentUser) {
			this.setState({
				moreOptionsVisible: false,
			});
			this.props.discussionCommentReport(discussionComment._id).catch((e) => {
				console.log(e);
			});
		}
	}
	unreportDiscussionComment() {
		const { currentUser, discussionComment } = this.props;
		if (currentUser) {
			this.setState({
				readComment: false,
			});
			this.props.discussionCommentUnreport(discussionComment._id).catch((e) => {
				console.log(e);
			});
		}
	}
	toggleMoreOptions() {
		this.setState({
			moreOptionsVisible: !this.state.moreOptionsVisible,
			shareOptionsVisible: false,
		});
	}
	toggleShareOptions() {
		this.setState({
			shareOptionsVisible: !this.state.shareOptionsVisible,
			moreOptionsVisible: false,
		});
	}
	readDiscussionComment() {
		this.setState({
			readComment: true,
		});
	}
	render() {
		const self = this;
		const userIsLoggedIn = Meteor.user();
		const { discussionComment, user } = this.props;
		let userLink = '';
		let userUpvoted = false;
		let userReported = false;
		let username = '';
		let offsetLeft = 0;
		let status;

		if (!discussionComment) {
			return null;
		}

		// Child discussion Comments
		discussionComment.children = [];

		// Make user link and user name
		if (user) {
			if (user.username) {
				username = user.username;
				userLink = `/users/${user._id}/${user.username}`;
			} else if (
				'emails' in user
				&& user.emails.length
			) {
				userLink = `/users/${user._id}`;
				username = user.emails[0].address.split('@')[0];
			}
		}

		if (
			this.props.currentUser &&
			discussionComment.voters &&
			discussionComment.voters.indexOf(this.props.currentUser._id) >= 0
		) {
			userUpvoted = true;
		}

		if (
			this.props.currentUser &&
			discussionComment.usersReported &&
			discussionComment.reported > 0 &&
			discussionComment.usersReported.indexOf(this.props.currentUser._id) >= 0
		) {
			userReported = true;
		}

		if (this.state.moreOptionsVisible) {
			offsetLeft = $('.toggle-more-button').position().left;
		}

		// Make status message if applicable
		if (discussionComment.status === 'pending') {
			status = 'Pending approval';
		} else if (discussionComment.status === 'trash') {
			status = 'This comment was made private by an Administrator';
		}

		let avatarUrl = '/images/default_user.jpg';
		if (user && user.profile && user.profile.avatarUrl) {
			avatarUrl = user.profile.avatarUrl;
		}

		return (
			<div className={`discussion-comment paper-shadow ${(userReported && !this.state.readComment ? 'discussion-comment--user-reported' : '')}`}>
				{status ?
					<span className="discussion-comment-status">
						{status}
					</span>
				: ''}
				<div className="discussion-comment-content">
					<div className="inner-comment-row">
						<div className="discussion-commenter-profile-picture profile-picture paper-shadow">
							<Link to={userLink}>
								<img
									src={avatarUrl}
									alt={username}
								/>
							</Link>
						</div>

						<div className="discussion-commenter-meta">
							<Link to={userLink}>
								<span className="discussion-commenter-name">
									{username}
								</span>
							</Link>
							<span className="discussion-comment-date">
								<span>{discussionComment.updated ? 'Updated: ' : 'Created: '}</span>
								{moment(discussionComment.updated ||
									discussionComment.created).format('D MMMM YYYY')}
							</span>
						</div>

					</div>
					<div className="inner-comment-row">
						<div className="discussion-comment-text">
							{this.state.editMode ?
								<form
									className="update-comment-form clearfix"
									name="update-comment-form"
									ref={(component) => { this.updateCommentForm = component; }}
								>
									<textarea
										className="new-comment-text"
										defaultValue={this.props.discussionComment.content}
									/>
									<div className="comment-edit-buttons">
										<RaisedButton
											label="Update"
											className="submit-comment-button paper-shadow"
											onClick={this.updateDiscussionComment}
										/>
										<FlatButton
											label="Close"
											className="close-form-button"
											onClick={this.closeEditMode}
										/>
									</div>
								</form>
								:
								<div>
									{discussionComment.content}
								</div>
							}
						</div>
					</div>
					<div className="inner-comment-row">
						{this.state.readComment === false ?
							<FlatButton
								label={discussionComment.votes}
								onClick={this.upvoteDiscussionComment}
								className={`discussion-comment-button vote-up ${(userUpvoted) ? 'upvoted' : ''}`}
								icon={<FontIcon className="mdi mdi-chevron-up" />}
								disabled={userUpvoted}
							>
								{!userIsLoggedIn ?
									<span className="md-tooltip">You must be signed in to vote.</span>
									:
									''
								}
							</FlatButton>
							:
							''
						}
						{(
								'currentUser' in self.props
							&& self.props.currentUser
							&& user
							&& self.props.currentUser._id === user._id
						) ?
							<FlatButton
								label="Edit"
								onClick={this.showEditMode}
								className="discussion-comment-button edit"
							/>
						:
							''
						}
						{this.state.readComment === false ?
							<FlatButton
								onClick={this.toggleMoreOptions}
								label=""
								className={`discussion-comment-button toggle-more-button ${(this.state.moreOptionsVisible) ? 'toggle-more-button--active' : ''}`}
								icon={<FontIcon className="mdi mdi-dots-horizontal" />}
							>
								<span className="md-tooltip">Show more</span>
							</FlatButton>
							:
							''
						}

						<div
							className={`more-options ${this.state.moreOptionsVisible ? 'more-options--visible' : ''}`}
							style={{
								left: offsetLeft,
							}}
						>
							<FlatButton
								label="Report"
								onClick={this.reportDiscussionComment}
								className={`discussion-comment-button report ${(userReported) ? 'reported' : ''}`}
								icon={<FontIcon className="mdi mdi-flag" />}
							>
								{!userIsLoggedIn ?
									<span className="md-tooltip">
										You must be signed in to report a comment.
									</span>
									:
									''
								}
							</FlatButton>
						</div>
						{this.state.readComment === true ?
							<FlatButton
								label="Unreport"
								onClick={this.unreportDiscussionComment}
							/>
							: ''
						}
					</div>

					<div className="discussion-comment-children">

						{discussionComment.children.map((discussionCommentChild, j) =>
							(<div
								key={j}
								className="discussion-comment discussion-comment-child"
							>
								<div className="inner-comment-row">
									<div className="discussion-commenter-profile-picture profile-picture paper-shadow">
										<img src="/images/default_user.png" alt={username} />
									</div>
									<div className="discussion-commenter-meta">
										<span className="discussion-commenter-name">
											{discussionCommentChild.user.name}
										</span>
										<span className="discussion-comment-date">
											{discussionCommentChild.updated}
										</span>
									</div>
								</div>
								<div className="inner-comment-row">
									<div className="discussion-comment-text">
										<p
											dangerouslySetInnerHTML={{
												__html: discussionCommentChild.content,
											}}
										/>
									</div>
								</div>
								<div className="inner-comment-row">
									<FlatButton
										label={discussionComment.votes}
										onClick={this.upvoteDiscussionComment}
										className="vote-up upvoted"
										icon={<FontIcon className="mdi mdi-chevron-up" />}
									/>
									<FlatButton
										label="Reply"
										onClick={this.showReplyForm}
										className="reply"
									/>
									<FlatButton
										label="Edit"
										onClick={this.editDiscussionComment}
										className="edit"
									/>
									<FlatButton
										label="Remove"
										onClick={this.removeDiscussionComment}
										className="remove"
									/>
								</div>

								{/* <!-- .discussion-comment-child --> */}
							</div>)
						)}
						{/* <!-- .discussion-comment-children --> */}
					</div>
					{/* <!-- .discussion-comment --> */}
				</div>
				<div className="discussion-comment-user-reported-message">
					You reported this comment.
					<div>
						<FlatButton
							label="Read"
							onClick={this.readDiscussionComment}
						/>
						<FlatButton
							label="Unreport"
							onClick={this.unreportDiscussionComment}
						/>
					</div>
				</div>
			</div>
		);
	}

}
DiscussionComment.propTypes = {
	discussionComment: PropTypes.object.isRequired,
	currentUser: PropTypes.object,
	user: PropTypes.object,
	discussionCommentUpdate: PropTypes.func,
	discussionCommentUnreport: PropTypes.func,
	discussionCommentReport: PropTypes.func,
	discussionCommentUpvote: PropTypes.func,



};
const cont = createContainer(({ discussionComment }) => {

	let handle;
	let user;

	if (discussionComment) {
		user = Meteor.users.findOne({ _id: discussionComment.userId });
	}

	return {
		user
	};

}, DiscussionComment);
export default compose(
	discussionCommentUpdateMutation, 
	discussionCommentReportMutation,
	discussionCommentsQuery,
	discussionCommentUpvoteMutation,
	discussionCommentUnreportMutation)(cont);
