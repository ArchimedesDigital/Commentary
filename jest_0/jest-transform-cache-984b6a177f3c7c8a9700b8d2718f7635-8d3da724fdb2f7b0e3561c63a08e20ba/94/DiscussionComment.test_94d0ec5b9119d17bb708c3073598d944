/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var /* istanbul ignore next */_MuiThemeProvider = require('material-ui/styles/MuiThemeProvider'); /* istanbul ignore next */var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);


var /* istanbul ignore next */_DiscussionComment = require('./DiscussionComment'); /* istanbul ignore next */var _DiscussionComment2 = _interopRequireDefault(_DiscussionComment);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('DiscussionComment', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_MuiThemeProvider2.default, /* istanbul ignore next */null,
			/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DiscussionComment2.default, /* istanbul ignore next */{
				discussionComment: {
					userId: 'testId',
					commentId: 'testId',
					comment: {
						work: {
							slug: 'iliad' },

						subwork: {
							title: '1' },

						lineFrom: 1 },

					status: 'publish',
					content: 'Quid faciat laetas segetes quo sidere terram vertere',
					otherCommentsCount: 0 } }))).




		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: