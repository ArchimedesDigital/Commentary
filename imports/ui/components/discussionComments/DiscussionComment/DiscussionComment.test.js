import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import DiscussionComment from './DiscussionComment';

describe('DiscussionComment', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<DiscussionComment
						discussionComment={{
							userId: 'testId',
							commentId: 'testId',
							comment: {
								work: {
									slug: 'iliad',
								},
								subwork: {
									title: '1'
								},
								lineFrom: 1,
							},
							status: 'publish',
							content: 'Quid faciat laetas segetes quo sidere terram vertere',
							otherCommentsCount: 0,
						}}
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
