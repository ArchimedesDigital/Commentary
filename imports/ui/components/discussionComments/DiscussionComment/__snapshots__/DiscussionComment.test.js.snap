// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionComment renders correctly 1`] = `
<div
  className="discussion-comment paper-shadow "
>
  
  <div
    className="discussion-comment-content"
  >
    <div
      className="inner-comment-row"
    >
      <div
        className="discussion-commenter-profile-picture profile-picture paper-shadow"
      >
        <a
          href=""
        >
          <img
            alt=""
            src="/images/default_user.jpg"
          />
        </a>
      </div>
      <div
        className="discussion-commenter-meta"
      >
        <a
          href=""
        >
          <span
            className="discussion-commenter-name"
          >
            
          </span>
        </a>
        <span
          className="discussion-comment-date"
        >
          <span>
            Created: 
          </span>
        </span>
      </div>
    </div>
    <div
      className="inner-comment-row"
    >
      <div
        className="discussion-comment-text"
      >
        <div>
          Quid faciat laetas segetes quo sidere terram vertere
        </div>
      </div>
    </div>
    <div
      className="inner-comment-row"
    >
      <button
        className="discussion-comment-button vote-up "
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="md-tooltip"
          >
            You must be signed in to vote.
          </span>
          <span
            className="mdi mdi-chevron-up"
            color="rgba(0, 0, 0, 0.87)"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
              }
            }
          />
        </div>
      </button>
      
      <button
        className="discussion-comment-button toggle-more-button "
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="md-tooltip"
          >
            Show more
          </span>
          <span
            className="mdi mdi-dots-horizontal"
            color="rgba(0, 0, 0, 0.87)"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
              }
            }
          />
        </div>
      </button>
      <div
        className="more-options "
        style={
          Object {
            "left": 0,
          }
        }
      >
        <button
          className="discussion-comment-button report "
          disabled={false}
          href={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transform": "translate(0, 0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": null,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="md-tooltip"
            >
              You must be signed in to report a comment.
            </span>
            <span
              className="mdi mdi-flag"
              color="rgba(0, 0, 0, 0.87)"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fontSize": 24,
                  "marginLeft": 12,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                }
              }
            />
            <span
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "muiPrepared": true,
                  "paddingLeft": 8,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "verticalAlign": "middle",
                }
              }
            >
              Report
            </span>
          </div>
        </button>
      </div>
      
    </div>
    <div
      className="discussion-comment-children"
    />
  </div>
  <div
    className="discussion-comment-user-reported-message"
  >
    You reported this comment.
    <div>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Read
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Unreport
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
